### PR TITLE
Mots 364: Front-end and Back-end validation to day of birth.

### PR DESCRIPTION
**Pull request by pstudzinski**
Front-end and Back-end validation to day of birth.
https://applab.atlassian.net/browse/MOTS-364

### CODE REVIEW

**Comment by dserkowski** [comment]
How about mobile application validation?

File: src/main/java/org/motechproject/mots/validate/constraintvalidators/PastDateFormatValidator.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  @Override
+  public boolean isValid(String value, ConstraintValidatorContext context) {
+    if (StringUtils.isEmpty(value)
+        || isValidDateFormat(MotsConstants.SIMPLE_DATE_FORMAT, value)) {
+      SimpleDateFormat sdf = new SimpleDateFormat(MotsConstants.SIMPLE_DATE_FORMAT);
+      try {
+        Date date = sdf.parse(value);
+        return date.before(new Date());
+      } catch (ParseException ex) {
+        return true;
```
Comment (kkaczmarczyk) [inline src/main/java/org/motechproject/mots/validate/constraintvalidators/PastDateFormatValidator.java]: should we return false here?

Comment (pstudzinski) [reply src/main/java/org/motechproject/mots/validate/constraintvalidators/PastDateFormatValidator.java]: We use PastDate annotation to check if date is in past. To check format we use DateFormat annotation.

**Comment by pstudzinski** [comment]
Added day of birth validation to Mobile App.

File: mobile/components/HealthWorkersForm.js [hunk 2]
```diff
@@ -260,6 +260,7 @@ class HealthWorkersForm extends Component {
         districtId={this.props.districtId}
         chiefdomId={this.props.chiefdomId}
         facilityId={this.props.facilityId}
+        dateOfBirth={this.props.dateOfBirth}
```
Comment (pmuchowski) [inline mobile/components/HealthWorkersForm.js]: why did you added it here?

File: mobile/components/HealthWorkersForm.js [hunk 3]
```diff
@@ -328,6 +336,7 @@ function mapStateToProps(state) {
     districtId: selector(state, 'districtId'),
     chiefdomId: selector(state, 'chiefdomId'),
     facilityId: selector(state, 'facilityId'),
+    dateOfBirth: selector(state, 'dateOfBirth'),
```
Comment (pmuchowski) [inline mobile/components/HealthWorkersForm.js]: you can remove the selector too

File: src/webapp/js/components/health-workers-form.js [hunk 4]
```diff
@@ -226,6 +233,7 @@ function mapStateToProps(state) {
     districtId: selector(state, 'districtId'),
     chiefdomId: selector(state, 'chiefdomId'),
     facilityId: selector(state, 'facilityId'),
+    dateOfBirth: selector(state, 'dateOfBirth'),
```
Comment (pmuchowski) [inline src/webapp/js/components/health-workers-form.js]: as above

**Pull request merged**